ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,199 @@
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        """get the prometheus query result
+
+        Args:
+            query(str): promthetheus query string
+
+        Returns:
+            dict: query result
+        """
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_alert(self, alert_name):
+        """
+        Get alert by alert anem if it's an active alert
+
+        Args:
+             alert(str): alert name
+
+        Examples:
+             result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             dict: dictiionary containing alert name as key and alert data as value
+        """
+        response = self.alerts
+        LOGGER.info(f" response activer alerts : {response}")
+        matched_alerts = {}
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert and alert["labels"]["alertname"] == alert_name:
+                alert_list.append(alert)
+        if not alert_list:
+            LOGGER.error(f"No Matching alerts found for alert: {alert_name}")
+        else:
+            matched_alerts[alert_name] = alert_list
+            return matched_alerts
+
+    def alert_sampler(self, alert, timeout):
+        """
+        Sample output for an alert and raises timeout exception
+
+        Args:
+             alert(str): alert name
+             timeout(int): wait time, default is 10 mins
+
+        Raise:
+             TimeoutExpiredError: if alert is not fired before wait_timeout
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout or TIMEOUT_10MIN,
+            sleep=self.scrapeInterval,
+            func=self.get_alert,
+            alert_name=alert,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                if sample:
+                    for item in sample[alert]:
+                        if item and item["state"] == "firing":
+                            return
+                else:
+                    LOGGER.error("no alert found")
+        except TimeoutExpiredError:
+            LOGGER.error(f"Failed to get successful alert {alert}  ")
+            raise
+
+    def get_scrape_interval(self):
+        """
+        get prometheus scrap interval
+
+        Returns:
+             int : scrape time interval
+        """
+
+        response = self._get_response(query=f"{self.api_v1}/targets")
+        result = response["data"]["activeTargets"]
+        for item in result:
+            if item and item["labels"]["job"] == "prometheus-k8s":
+                LOGGER.info(f" item : {item}")
+                scrape_interval = item["scrapeInterval"]
+                scrape_interval = int((re.match(r"\d+", scrape_interval)).group())
+                break
+        return scrape_interval
```

Comment:
What if we don't find a matching item?

Reply:
doesn't that mean scrape interval is not set, can set the default to 30s

Reply:
yes, please.

Reply:
done